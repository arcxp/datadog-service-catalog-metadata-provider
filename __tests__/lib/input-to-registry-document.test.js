const YAML = require('yaml')
const core = require('@actions/core')

const Ajv = require('ajv')
const ddSchema = require('../data/datadog-service-catalog-schema.json')
const validate = new Ajv({ strict: false, validateFormats: false }).compile(
  ddSchema,
)

const { expandObjectInputs } = require('../../lib/input-expander')
const {
  inputsToRegistryDocument,
} = require('../../lib/input-to-registry-document')

describe('input-to-registry-document.js', () => {
  test('#inputsToRegistryDocument() - minimal input', async () => {
    const testInput = `
---
datadog-key: FAKE_KEY
datadog-app-key: FAKE_KEY
service-name: datadog-service-catalog-metadata-provider-test
team: Team Name Here
email: 'team-name-here@fakeemaildomainthatdoesntexist.com'
slack-support-channel: 'https://fakeorg.slack.com/archives/A0000000000'
repo: https://github.com/arcxp/datadog-service-catalog-metadata-provider
tags: |
  - 'application:GitHub Action Config Test'
  - env:prod
  - infrastructure:serverless
  - language:nodejs
repos: |
  - url: https://github.com/actions/toolkit
    provider: Github
    name: "@actions/toolkit"
links: |
  - name: AMI Version Status Dashboard
    url: https://thisisanentirelyfakeurl.seriouslythisisafakehostname.com/dashboard
    type: dashboard
docs: |
  - name: GitHub Actions!
    url: https://github.com/features/actions
    provider: Github
integrations: |
  opsgenie:
    service-url: https://yourorghere.app.opsgenie.com/service/00000000-0000-0000-0000-000000000000
    region: US
    `

    core.__setInputsObject(YAML.parse(testInput))
    const inputs = await inputsToRegistryDocument()
    expect(validate(inputs)).toStrictEqual(true)
    expect(validate.errors).toBeNull()
    expect(inputs).toMatchSnapshot()
  })
})
