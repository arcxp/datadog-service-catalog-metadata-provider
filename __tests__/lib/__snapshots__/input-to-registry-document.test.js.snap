// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input-to-registry-document.js #inputsToRegistryDocument() - less input 1`] = `
{
  "contacts": [
    {
      "contact": "team-name-here@fakeemaildomainthatdoesntexist.com",
      "name": "Team Email",
      "type": "email",
    },
    {
      "contact": "https://fakeorg.slack.com/archives/A0000000000",
      "name": "Support Slack Channel",
      "type": "slack",
    },
  ],
  "dd-service": "datadog-service-catalog-metadata-provider-test",
  "docs": [],
  "integrations": {
    "opsgenie": {
      "region": "US",
      "service-url": "https://yourorghere.app.opsgenie.com/service/00000000-0000-0000-0000-000000000000",
    },
  },
  "links": [],
  "repos": [
    {
      "name": "Service Repository",
      "provider": "Github",
      "url": "https://github.com/arcxp/datadog-service-catalog-metadata-provider",
    },
  ],
  "schema-version": "v2",
  "tags": [
    "application:GitHub Action Config Test",
    "env:prod",
    "infrastructure:serverless",
    "language:nodejs",
  ],
  "team": "Team Name Here",
}
`;

exports[`input-to-registry-document.js #inputsToRegistryDocument() - normal input 1`] = `
{
  "contacts": [
    {
      "contact": "team-name-here@fakeemaildomainthatdoesntexist.com",
      "name": "Team Email",
      "type": "email",
    },
    {
      "contact": "https://fakeorg.slack.com/archives/A0000000000",
      "name": "Support Slack Channel",
      "type": "slack",
    },
  ],
  "dd-service": "datadog-service-catalog-metadata-provider-test",
  "docs": [
    {
      "name": "GitHub Actions!",
      "provider": "Github",
      "url": "https://github.com/features/actions",
    },
  ],
  "integrations": {
    "opsgenie": {
      "region": "US",
      "service-url": "https://yourorghere.app.opsgenie.com/service/00000000-0000-0000-0000-000000000000",
    },
  },
  "links": [
    {
      "name": "AMI Version Status Dashboard",
      "type": "dashboard",
      "url": "https://thisisanentirelyfakeurl.seriouslythisisafakehostname.com/dashboard",
    },
  ],
  "repos": [
    {
      "name": "Service Repository",
      "provider": "Github",
      "url": "https://github.com/arcxp/datadog-service-catalog-metadata-provider",
    },
    {
      "name": "@actions/toolkit",
      "provider": "Github",
      "url": "https://github.com/actions/toolkit",
    },
  ],
  "schema-version": "v2",
  "tags": [
    "application:GitHub Action Config Test",
    "env:prod",
    "infrastructure:serverless",
    "language:nodejs",
    "other:value",
  ],
  "team": "Team Name Here",
}
`;
